core/arch/interp.c

```diff
@@ -3744,6 +3744,21 @@ build_bb_ilist(dcontext_t *dcontext, build_bb_t *bb)
             break;
         }
 
+#ifdef X86
+        if (my_dcontext != NULL && instr_is_syscall(bb->instr) &&
+            instr_get_opcode(bb->instr) == OP_int &&
```

Comment:
You can see that the UNIX case right above what you're quoting does relax it and has do_int_syscall vs the sysenter do_syscall.  We can do something similar for Windows.